cibyl/models/ci/zuul/job.py

```diff
@@ -139,8 +139,14 @@ def __init__(self, name, url, variants=None, builds=None):
 
     @overrides
     def merge(self, other):
+        # Merging with oneself will never end, avoid at all costs.
```

Comment:
It might be worth to add to the comment that this infinite loop might happen because the variants are stored using and AttributeListValue and not a Dict

Reply:
True, but on any case merging with oneself does not make much sense.